### ISSUE TITLE
row limit settings are not clear

### ISSUE THREAD

**Issue by tooptoop4** [apache/incubator-superset#9947]
```python
ROW_LIMIT = 50000
VIZ_ROW_LIMIT = 10000
FILTER_SELECT_ROW_LIMIT = 10000
QUERY_SEARCH_LIMIT = 1000
SQL_MAX_ROW = 10000000
DISPLAY_MAX_ROW = 10000
DEFAULT_SQLLAB_LIMIT = 10000
```

**Comment by mistercrunch** [comment]
Agreed, this needs to get cleaned up and better documented.

**Comment by tooptoop4** [comment]
can this be reopened/pinned @mistercrunch ?

**Comment by john-bodley** [comment]
I agree this needs to be cleaned up. The various limits (especially in SQL Lab) seem overly complex and it's not apparent to the user when which limits apply. Furthermore it's not clear to the user whether the rendered result set is further truncated (beyond the injected limit in the executed SQL) and/or how these limits apply when either exploring or downloading the dataset. 

Granted improved documentation would help, but I sense simplifying the logic would be more desirable. I wonder if we can glean some insights in how other SQL IDEs define and implement limits.

@nytai are you aware if Preset et al. are actively thinking about this?

**Comment by nytai** [comment]
@betodealmeida or @eschutho would know better as their team recently did some work in this area

**Comment by eschutho** [comment]
cc @yousoph. Yes, we're making some improvements on the user side when they view their results to understand why it was limited, but agree that better documentation on the admin side or simplifying/condensing some of the values could help as well. We should have something up next week for more user messaging.
